pkg/plugin/plugin_test.go

```diff
@@ -350,6 +350,18 @@ func TestValidatePluginData(t *testing.T) {
 	}
 }
 
+func TestValidatePluginData_Empty_Plugin(t *testing.T) {
+	err := validatePluginData(nil, "")
+	if err == nil {
+		t.Error("should have trapped nil pointer for empty plugin")
+	}
+	p := new(Plugin)
+	err = validatePluginData(p, "")
```

Comment:
This can also be one-lined: `if err := validatePluginData(new(Plugin), “”); err == nil {`

There is no need to assign a new variable `p` when it will not be reused more than once.